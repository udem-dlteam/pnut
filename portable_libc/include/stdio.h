#ifndef _STDIO_H
#define _STDIO_H

#include "sys/types.h"
#include "stdarg.h"

#ifdef USE_STRUCT

typedef struct {
  int fd;
  char buf[1];
  char *string_output_buf;
  size_t string_output_buf_size;
  size_t string_output_len;
} FILE;

#else

typedef int FILE;

#endif

#ifdef PNUT_CC

FILE *stdin;
FILE *stdout;
FILE *stderr;

#else

extern FILE *stdin;
extern FILE *stdout;
extern FILE *stderr;

#endif

FILE *fopen(const char *pathname, const char *mode);
FILE *fdopen(int fd, const char *mode);
int fclose(FILE *stream);

int fputc(int c, FILE *stream);
size_t fwrite(const void *ptr, size_t size, size_t nmemb, FILE *stream);
int fputs(const char *s, FILE *stream);
int puts(const char *s);

size_t fread(void *buffer, size_t size, size_t count, FILE *stream);
int fseek( FILE* stream, long offset, int origin );
long ftell( FILE* stream );
int remove(const char *_Filename);

int vfprintf(FILE *stream, const char *format, va_list ap);
int fprintf(FILE *stream, const char *format VAR_ARGS);
int printf(const char *format VAR_ARGS);

int vsnprintf(char *str, size_t size, const char *format, va_list ap);
int snprintf(char *str, size_t size, const char *format VAR_ARGS);
int sprintf(char *str, const char *format VAR_ARGS);

#endif
