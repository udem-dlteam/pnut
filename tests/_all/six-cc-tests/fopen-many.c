// On mksh, this test takes more than 30s. Adjusting the timeout to make it pass.
// timeout: 120

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  int i = 0;
  while (i < 100) {
    f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
    while ((c = fgetc(f)) != -1) {
      putchar(c);
    }
    putchar('\n');
    fclose(f);
    i = i + 1;
  }
}
