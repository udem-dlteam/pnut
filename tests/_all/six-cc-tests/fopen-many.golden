#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

#include <stdio.h>

#ifdef PNUT_CC
typedef int FILE;
#endif

int main() {
  FILE *f;
  char c;
  f = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  while ((c = fgetc(f)) != -1) {
    putchar(c);
  }
  fclose(f);

  return 0;
}

